app/controllers/authenticated_controller.rb

```diff
@@ -47,5 +42,19 @@ def user_id
     current_user.id
   end
 
+  def check_roles!
+    unless allowed?
+      redirect_to root_path, alert: "Your user <strong>#{current_user&.email_address}</strong> does not have the required permissions to access this section"
+    end
+  end
+
+  def allowed_roles
+    ["ott:full"]
+  end
+
+  def allowed?
+    allowed_roles.any? && allowed_roles.any? { |role| organisation&.has_role?(role) }
```

Comment:
I think the logic has changed in the last commit. Should it be this?
```suggestion
    allowed_roles.none? || allowed_roles.any? { |role| organisation&.has_role?(role) }
```